### PR TITLE
Fix Discord & Google Calendar UUIDs

### PR DESCRIPTION
**Pull request by nik-dange**
# Description

What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context.

Made Discord UUIDs and Google Calendar UUIDs strings because they aren't V4 UUIDs

## Changes

- See above

# Type of Change

- [x] Patch (non-breaking change/bugfix)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Documentation (A change to a README/description)
- [ ] Continuous Integration/DevOps Change (Related to deployment steps, continuous integration
      workflows, linting, etc.)
- [ ] Other: (Fill In) 

If you've selected Patch, Minor, or Major as your change type, **make sure to bump the version before merging in `package.json`!**
# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally.
- [ ] on the testing API/testing database.
- [ ] with appropriate Postman routes. Screenshots are included below.

# Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have appropriately edited the API version in the `package.json` file.
- [ ] My changes produce no new warnings.

# Screenshots

Please include a screenshot of your Postman testing passing successfully.

### CODE REVIEW

File: models/EventModel.ts [hunk 1]
```diff
@@ -67,11 +67,11 @@ export class EventModel extends BaseEntity {
   @OneToMany((type) => ExpressCheckinModel, (expressCheckin) => expressCheckin.event, { cascade: true })
   expressCheckins: ExpressCheckinModel[];
 
-  @Column('uuid', { nullable: true })
-  discordEvent: Uuid;
+  @Column('varchar', { nullable: true })
+  discordEvent: string;
```
Comment (dowhep) [inline models/EventModel.ts:71]: In types/index.ts, we define Uuid with `type Uuid = string` and we just use it for clarification on the context we use it. As a result, I think it is better to leave Uuid as type here as well

File: tests/data/EventFactory.ts [hunk 2]
```diff
@@ -41,8 +41,8 @@ export class EventFactory {
       deleted: false,
       eventLink: faker.internet.url(),
       thumbnail: FactoryUtils.getRandomImageUrl(),
-      discordEvent: uuid(),
-      googleCalendarEvent: uuid(),
+      discordEvent: faker.datatype.hexaDecimal(10),
```
Comment (dowhep) [inline tests/data/EventFactory.ts:44]: Is this the way discord generate Uuid or is this just random string?

File: types/ApiRequests.ts [hunk 3]
```diff
@@ -155,8 +155,8 @@ export interface OptionalEventProperties {
   eventLink?: string;
   requiresStaff?: boolean;
   staffPointBonus?: number;
-  discordEvent?: Uuid;
-  googleCalendarEvent?: Uuid;
+  discordEvent?: string;
```
Comment (dowhep) [inline types/ApiRequests.ts:158]: Same with the Uuid comment above

**Review by dowhep — CHANGES_REQUESTED** [review]
nit on change on uuid

File: tests/data/EventFactory.ts (hunk 2 above)
Comment (nik-dange) [reply tests/data/EventFactory.ts:44]: This is just a random string. Discord UUIDs are different (see this error message: https://discord.com/channels/573028991527550986/1012204566122598431/1236042800043397261) than what Postgres allows for UUIDs (https://www.postgresql.org/docs/current/datatype-uuid.html), which is why I opted to make it a string

File: models/EventModel.ts (hunk 1 above)
Comment (nik-dange) [reply models/EventModel.ts:71]: So to clarify, the column type can be `varchar`, but it should be `discordEvent: Uuid`?

File: migrations/0045-fix-discord-gcal-field.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class FixDiscordGcalField1714770061929 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query('ALTER TABLE "Events" ALTER COLUMN "discordEvent" TYPE varchar');
+    await queryRunner.query('ALTER TABLE "Events" ALTER COLUMN "googleCalendarEvent" TYPE varchar');
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    // nothing here because it's fixing an earlier migration (# 0044)
```
Comment (dowhep) [inline migrations/0045-fix-discord-gcal-field.ts:10]: Despite this being a fix, I believe it is still a good practice to alter the table type back to what it was originally.

**Review by dowhep — CHANGES_REQUESTED** [review]
One suggested change regarding migration

**Review by dowhep — APPROVED** [review]
LFTM 🚀 🚀 

Upon discussion, the down migration is deemed not necessary and will create more problems to implement than to just leave it.

**Pull request merged**